src/main/scala/rules/MagicWandSupporter.scala

```diff
@@ -250,67 +227,93 @@ object magicWandSupporter extends SymbolicExecutionRules {
 
-      def appendToResults(s5: State, ch: Chunk, pcs: RecordedPathConditions, v4: Verifier): Unit = {
-        assert(s5.conservedPcs.nonEmpty, s"Unexpected structure of s5.conservedPcs: ${s5.conservedPcs}")
+      var conservedPcsStack: Stack[Vector[RecordedPathConditions]] = s5.conservedPcs
 
-        var conservedPcs: Vector[RecordedPathConditions] = Vector.empty
-        var conservedPcsStack: Stack[Vector[RecordedPathConditions]] = s5.conservedPcs
+      // Producing a wand's LHS and executing the packaging proof code can introduce definitional path conditions, e.g.
+      // new permission and snapshot maps, which are in general necessary to proceed after the
+      // package statement, e.g. to know which permissions have been consumed.
+      // Here, we want to keep *only* the definitions, but no other path conditions.
 
-        // Producing a wand's LHS and executing the packaging proof code can introduce definitional path conditions, e.g.
-        // new permission and snapshot maps, which are in general necessary to proceed after the
-        // package statement, e.g. to know which permissions have been consumed.
-        // Here, we want to keep *only* the definitions, but no other path conditions.
+      conservedPcsStack =
+        s5.conservedPcs.tail match {
+          case empty @ Seq() => empty
+          case head +: tail => (head ++ (s5.conservedPcs.head :+ pcs.definitionsOnly)) +: tail
+        }
 
-        conservedPcs = s5.conservedPcs.head :+ pcs.definitionsOnly
+      val s6 = s5.copy(conservedPcs = conservedPcsStack, recordPcs = s.recordPcs)
 
-        conservedPcsStack =
-          s5.conservedPcs.tail match {
-            case empty @ Seq() => empty
-            case head +: tail => (head ++ conservedPcs) +: tail
-          }
+      recordedBranches :+= (s6, v4.decider.pcs.branchConditions, v4.decider.pcs.branchConditionExps, conservedPcs, ch)
+    }
 
-        val s6 = s5.copy(conservedPcs = conservedPcsStack, recordPcs = s.recordPcs)
+    def createWandChunkAndRecordResults(s4: State,
+                                        freshSnapRoot: Var,
+                                        snapRhs: Term,
+                                        v3: Verifier)
+                                       : VerificationResult = {
+      val preMark = v3.decider.setPathConditionMark()
 
-        results :+= (s6, v4.decider.pcs.branchConditions, v4.decider.pcs.branchConditionExps, conservedPcs, ch)
+      v3.decider.prover.comment(s"Create WandMap for wand $wand")
+      val wandSnapshot: MagicWandSnapshot = if (useMWSF(s4.program)) {
+        val fct = MagicWandSnapFunction(freshSnapRoot, snapRhs, v3.decider.fresh("mwsf", sorts.MagicWandSnapFunction))
+        v3.decider.assumeDefinition(fct.definition)
+        fct
+      } else {
+        MagicWandSnapSingleton(freshSnapRoot, snapRhs)
       }
 
-      val preMark = v3.decider.setPathConditionMark()
+      // If the wand is part of a quantified expression
```

Comment:
No, this checks if this particular want type is used as a quantified resource anywhere in the current function/method, and thus whether chunks for this wand use the internal representation for quantified resources.